src/db.rs

```diff
@@ -73,15 +74,19 @@ impl Storage for DB {
             )
         };
         unsafe {
-            go_result.into_ffi_result(err, default)?;
+            if let Err(err) = go_result.into_ffi_result(err, default) {
+                return (Err(err), gas_info);
+            }
         }
 
         if result_buf.ptr.is_null() {
-            return Ok((None, used_gas));
+            return (Ok(None), gas_info);
         }
         // We initialize `result_buf` with a null pointer. if it is not null,
         // that means it was initialized by the go code, with values generated by `memory::allocate_rust`
-        Ok((unsafe { Some(result_buf.consume()) }, 0))
+        let data = unsafe { result_buf.consume() };
+        // TODO: why is this 0?
+        (Ok(Some(data)), GasInfo::with_externally_used(0))
```

Comment:
Remaining discussion:

> This line actually does look like a bug!
> gas usage was not returned when the value was returned correctly!

from @reuvenpo

Reply:
Yes, we should return `gas_info` like everywhere else

Reply:
Done in 5aa13ec7b3baa12a1fe5b2363fdcdf6352b4009b